### ISSUE TITLE
`prefer_key_path` rule should not trigger on macros

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#5744]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

The rule triggers on this, but it cannot use a key path.

```swift
#Predicate { $0.isCompleted }
```

### Environment

* SwiftLint version: 0.56.0
* Xcode version: Xcode 15.4, Build version 15F31d
* Installation method used: Installer
* Configuration file:

```yml
only_rules:
  - prefer_key_path
```
